File: sortAlgorithms.cs
using System.Runtime.CompilerServices;


Console.WriteLine("Si inserisca la quantità di valori da ordinare: ");
int n = int.Parse(Console.ReadLine());
int[] array = new int[n];
for (int i = 0; i < n; i++)
{
    Console.WriteLine("Inserire il valore " + (i + 1) + ": ");
    array[i] = int.Parse(Console.ReadLine());
}
Console.WriteLine("Si inserisca su quale algoritmo si vuole ordinare: ");
Console.WriteLine("1. Bubble Sort \n 2. Insertion Sort \n 3. Selection Sort");
int scelta = int.Parse(Console.ReadLine());
switch (scelta)
{
    case 1:
        bubbleSort(array);
        break;
    case 2:
        insertionSort(array);
        break;
    case 3:
        selectionSort(array);
        break;
    default:
        Console.WriteLine("Scelta non valida");
        break;
}
bubbleSort(arrayBubble);
Console.WriteLine("\nArray after bubble sorting:");
foreach (int i in arrayBubble)
{
    Console.Write(i + " ");
}




//metodi
static void bubbleSort(int[] array)
{
    //temporanea variabile
    int temp;
    for (int i = 0; i < array.Length; i++)
    {
        for (int j = 0; j < array.Length - 1; j++)
        {
            //se array[j] è maggiore di array[j + 1]
            if (array[j] > array[j + 1])
            {
                //temp ottiene il valore di array[j + 1]
                temp = array[j + 1];
                //array[j + 1] ottiene il valore di array[j]
                array[j + 1] = array[j];
                //array[j] ottiene il valore di temp
                array[j] = temp;
            }
        }
    }
}

static void insertionSort(int[] array)
{
    //key è la variabile temporanea
    int key, y;
    for (int i = 1; i < array.Length; i++)
    {
        //key ottiene il valore di array[i]
        key = array[i];
        //y ottiene il valore di i - 1
        y = i - 1;
        //finchè y è maggiore o uguale a 0 e array[y] è maggiore di key
        while (y >= 0 && array[y] > key)
        {
            //array[y + 1] ottiene il valore di array[y]
            array[y + 1] = array[y];
            //y ottiene il valore di y - 1
            y = y - 1;
        }
        //array[y + 1] ottiene il valore d[y + 1] = key;
    }
}

static void selectionSort(int[] array)
{
    //minIndex è la variabile temporanea
    int minIndex, temp1;
    for (int i = 0; i < array.Length - 1; i++)
    {
        //minIndex ottiene il valore di i
        minIndex = i;
        for (int j = i + 1; j < array.Length; j++)
        {
            //se array[j] è minore di array[minIndex]
            if (array[j] < array[minIndex])
            {
                //minIndex ottiene il valore di j
                minIndex = j;
            }
        }
        //temp1 ottiene il valore di array[i]
        temp1 = array[i];
        //array[i] ottiene il valore di array[minIndex]
        array[i] = array[minIndex];
        //array[minIndex] ottiene il valore di temp1
        array[minIndex] = temp1;
    }
}

File: sortAlgorithms.cpp
using namespace std;
#include <iostream>

void bubbleSort(int array[], int n);
void insertionSort(int array[], int n);
void selectionSort(int array[], int n);

int main() {
    cout << "Si inserisca la quantità di valori da ordinare: ";
    int n;
    cin >> n;
    int array[n];
    for (int i = 0; i < n; i++) {
        cout << "Inserire il valore " << (i + 1) << ": ";
        cin >> array[i];
    }
    cout << "Si inserisca su quale algoritmo si vuole ordinare: " << endl;
    cout << "1. Bubble Sort" << endl;
    cout << "2. Insertion Sort" << endl;
    cout << "3. Selection Sort" << endl;
    int scelta;
    cin >> scelta;
    switch (scelta) {
        case 1:
            bubbleSort(array, n);
            break;
        case 2:
            insertionSort(array, n);
            break;
        case 3:
            selectionSort(array, n);
            break;
        default:
            cout << "Scelta non valida" << endl;
            break;
    }
    cout << "\nArray after sorting:" << endl;
    for (int i = 0; i < n; i++) {
        cout << array[i] << " ";
    }
    cout << endl;
    return 0;
}

void bubbleSort(int array[], int n) {
    int temp;
    for (int i = 0; i < n; i++) {
        for (int j = 0; j < n - 1; j++) {
            if (array[j] > array[j + 1]) {
                temp = array[j + 1];
                array[j + 1] = array[j];
                array[j] = temp;
            }
        }
    }
}

void insertionSort(int array[], int n) {
    int key, y;
    for (int i = 1; i < n; i++) {
        key = array[i];
        y = i - 1;
        while (y >= 0 && array[y] > key) {
            array[y + 1] = array[y];
            y = y - 1;
        }
        array[y + 1] = key;
    }
}

void selectionSort(int array[], int n) {
    int minIndex, temp1;
    for (int i = 0; i < n - 1; i++) {
        minIndex = i;
        for (int j = i + 1; j < n; j++) {
            if (array[j] < array[minIndex]) {
                minIndex = j;
            }
        }
        temp1 = array[i];
        array[i] = array[minIndex];
        array[minIndex] = temp1;
    }
}